examples/directml/dolly_v2/config_dolly_v2.json

```diff
@@ -1,9 +1,11 @@
 {
     "input_model":{
-        "type": "OptimumModel",
```

Comment:
I agree that the current `OptimumModel` is redundant since it is just a pytorch model with hf_config. I was considering removing it too. 

But I think the `decoder_model` and `decoder_with_past` might be better as config parameters for `optimum_conversion`? In the context of a pytorch model, they don't make much sense.

Reply:
They are just _names_ for models. Didn't see a point in changing them. Mostly to avoid any failures upstream if there was something depending on these models being called that.

Reply:
Since optimummodel is being removed, any incompatibility failures would already be caused by it? 

So imo, if we are fixing this redundant model handler, it's better to fix the weird placement of the names too. The names in the pytorch model makes it seem like it has multiple components, but it doesn't. It's only used by the optimum conversion pass to select the onnx components.

Reply:
> Since optimummodel is being removed, any incompatibility failures would already be caused by it?

Not exactly because OptimumModel would have followed by OptimumMerge which expected the inputs named certain way. That second part has been replaced with `merge_decoders` which also expect the models be named the same way.

Reply:
What I meant is that any example that uses optimummodel or any optimum passes need to be updated anyways since they are all tied to each other. 
So, moving the components as a parameter for optimum conversion pass would not cause any additional breakage.

Reply:
Puled out the change to OptimumMerge & OptimumModelHandler. PR focus on introducing merge decoders for OnnxConversion only.